IsraelHG/solvd-test-automation
# Debug

# Update
This update includes 5 test cases that test Google's main features:

1. **testGoogleSearch**:
   - Open the Google home page.
   - Perform a search with the provided search term (e.g., "amazon").
   - Check if the Google search result page has loaded successfully.
   - Get the link URL from the search result page for the provided company name (e.g., "amazon").
   - Validate that the link URL matches the expected link URL for the company (e.g., "https://www.amazon.com/").
   - Validate that the company name is present in all search results.

2. **testGoogleImagePage**:
   - Open the Google home page.
   - Check if the Google home page has loaded successfully.
   - Click on the "Images" button to navigate to the Google Images page.
   - Check if the Google Images page has loaded successfully.
   - Read the description of the Google Images icon.
   - Validate that the description is "Google Images".

3. **testGoogleProductSpecs**:
   - Open the Google home page.
   - Check if the Google home page has loaded successfully.
   - Click on the "Store" button to navigate to the Google Store page.
   - Check if the Google Store page has loaded successfully.
   - Search for the first product ("pixel 7 pro") in the search bar.
   - Validate that the product name is "Pixel 7 Pro" and the price is "From $899".
   - Refresh the Google Store page to clear the search results.
   - Search for the second product ("pixel 7a") in the search bar.
   - Validate that the product name is "Pixel 7a" and the price is "From $499".

4. **testGoogleRegionSettings**:
   - Open the Google home page.
   - Check if the Google home page has loaded successfully.
   - Change the theme of the Google home page.
   - Check if the Google home page has loaded successfully with the new theme.
   - Open the menu and navigate to the Search Settings page.
   - Check if the Google Settings page has loaded successfully.
   - Select the region "Afghanistan" from the settings.
   - Validate that the selected region is displayed on the Google home page.

5. **testGoogleLanguageSettings**:
   - Open the Google home page.
   - Check if the Google home page has loaded successfully.
   - Change the theme of the Google home page.
   - Check if the Google home page has loaded successfully with the new theme.
   - Open the menu and navigate to the Search Settings page.
   - Check if the Google Settings page has loaded successfully.
   - Select the language "Italian" from the settings.
   - Validate that the selected language is displayed on the Google home page.
